src/invenio-search-js/invenioSearchJs.module.js

```diff
@@ -567,6 +568,56 @@
   }
 
   /**
+   * @namespace InvenioSearchFacetsCheckboxGroup
+   * @desc Handles checkbox changes
+   * @memberOf Directives
+   */
+  function invenioSearchFacetsCheckboxGroup() {
+
+    return {
+      scope: false,
+      link: link
+    };
+
+    ////////////
+
+    /**
+     * @name link
+     * @desc Synchronizes the controller with the directive
+     * @memberOf Controllers.invenioSearchBar
+     */
+    function link(scope, element, attrs) {
+      var key = scope.$parent.$parent.key;
+      var value = scope.item.key.trim();
+      var item = key + ':' + value;
+
+      // Check if the element is checked
+      var query = scope.invenioSearchQuery.split(' OR ');
+      if (query.indexOf(item) > 0) {
+        element[0].checked = true;
+      }
+
+      element.bind('click', function($event) {
```

Comment:
instead of building the query I would send the selected items as arguments - see hepdata.net